src/main/java/frc/robot/autonomous/TrajectoryFollower.java

```diff
@@ -0,0 +1,99 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+package frc.robot.autonomous;
+
+import com.google.errorprone.annotations.concurrent.GuardedBy;
+import edu.wpi.first.math.geometry.Pose2d;
+import edu.wpi.first.math.trajectory.Trajectory;
+import java.util.Optional;
+
+/** Follower for trajectory of type T */
+public abstract class TrajectoryFollower<T> {
+  private final Object trajectoryLock = new Object();
+
+  /** The trajectory that is currently being followed. null if no trajectory is being followed. */
+  @GuardedBy("trajectoryLock")
+  private Trajectory currentTrajectory = null;
+
+  /**
+   * The time that the current trajectory started to be followed. NaN if the trajectory has not been
+   * started yet.
+   */
+  @GuardedBy("trajectoryLock")
```

Comment:
If there are no threads, do we need guards?

Reply:
idt so